catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/StorageProviderImpl.java

```diff
@@ -127,44 +132,44 @@ public ReadStorageResponse read(ReadStorageRequest readRequest) throws StorageEx
         String id = uri.getSchemeSpecificPart();
         Path file = getContentFilePath(id);
 
-        if (file != null) {
-            String extension = FilenameUtils.getExtension(file.getFileName()
-                    .toString());
+        if (file == null) {
+            throw new StorageException("Unable to find file for content ID: " + id);
+        }
 
-            String mimeType;
+        String extension = FilenameUtils.getExtension(file.getFileName()
+                .toString());
 
-            try (InputStream fileInputStream = Files.newInputStream(file)) {
-                mimeType = mimeTypeMapper.guessMimeType(fileInputStream, extension);
-            } catch (Exception e) {
-                LOGGER.warn(
-                        "Could not determine mime type for file extension = {}; defaulting to {}",
-                        extension, DEFAULT_MIME_TYPE);
-                mimeType = DEFAULT_MIME_TYPE;
-            }
-            if (mimeType == null || DEFAULT_MIME_TYPE.equals(mimeType)) {
-                try {
-                    mimeType = Files.probeContentType(file);
-                } catch (IOException e) {
-                    LOGGER.debug("Unable to determine mime type using Java Files service.", e);
-                    mimeType = DEFAULT_MIME_TYPE;
-                }
-            }
+        String mimeType;
 
-            LOGGER.debug("mimeType = {}", mimeType);
-            long size = 0;
+        try (InputStream fileInputStream = Files.newInputStream(file)) {
+            mimeType = mimeTypeMapper.guessMimeType(fileInputStream, extension);
+        } catch (Exception e) {
+            LOGGER.warn("Could not determine mime type for file extension = {}; defaulting to {}",
+                    extension, DEFAULT_MIME_TYPE);
+            mimeType = DEFAULT_MIME_TYPE;
+        }
+        if (mimeType == null || DEFAULT_MIME_TYPE.equals(mimeType)) {
             try {
-                size = Files.size(file);
+                mimeType = Files.probeContentType(file);
             } catch (IOException e) {
-                LOGGER.warn("Unable to retrieve size of file.", e);
+                LOGGER.warn("Unable to determine mime type using Java Files service.", e);
+                mimeType = DEFAULT_MIME_TYPE;
             }
-            ContentItem returnItem = new ContentItemImpl(id,
-                    com.google.common.io.Files.asByteSource(file.toFile()), mimeType,
-                    file.toAbsolutePath()
-                            .toString(), size, null);
-            return new ReadStorageResponseImpl(readRequest, returnItem);
         }
 
-        throw new StorageException("Unable to find file for content ID: " + id);
+        LOGGER.debug("mimeType = {}", mimeType);
+        long size = 0;
+        try {
+            size = Files.size(file);
+        } catch (IOException e) {
+            LOGGER.warn("Unable to retrieve size of file.", e);
```

Comment:
Since this means we'll get extra WARN logs when the file is accessed (from `StorageProviderImpl` when it checks this size against the actual data), we might want to include the name of the file in the log so those messages can be correlated.